### PR TITLE
[ROAD-218] Feat: improved description panel header

### PR DESCRIPTION
**Pull request by aldanchenko**
Added common header for description panel (like in IJ).

Screenshot 2021-08-22 at 20 52 07

### CODE REVIEW

**Comment by michelkaporin** [comment]
@aldanchenko can you rebase this branch please, so that it's reviewable? Thank you 🙌

File: Snyk.VisualStudio.Extension/Snyk/VisualStudio/Extension/Service/SnykSolutionService.cs [hunk 1]
```diff
@@ -132,10 +132,10 @@ public string GetSolutionPath()
             {
                 this.logger.LogInformation("Get solution path from solution full name in case solution with projects.");
 
-                solutionPath = dteSolution.FullName;
+                //solutionPath = dteSolution.FullName;
```
Comment (michelkaporin) [inline Snyk.VisualStudio.Extension/Snyk/VisualStudio/Extension/Service/SnykSolutionService.cs:135]: Let's use comments to provide meaningful information for future code readers only 🙏

File: Snyk.VisualStudio.Extension/Snyk/VisualStudio/Extension/UI/Notifications/VsInfoBarService.cs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+        /// <summary>
+        /// Show message in infobar.
+        /// </summary>
+        /// <param name="message">Message.</param>
+        public void ShowWarningInfoBar(string message) => ThreadHelper.JoinableTaskFactory.Run(async () =>
+        {
+            await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+            InfoBarTextSpan text = new InfoBarTextSpan(message);
```
Comment (michelkaporin) [inline Snyk.VisualStudio.Extension/Snyk/VisualStudio/Extension/UI/Notifications/VsInfoBarService.cs:58]: Let's not fully type out class names and use `var`. It allows easier future refactorings and enhances readability.

File: Snyk.VisualStudio.Extension/Snyk/VisualStudio/Extension/UI/Toolwindow/DescriptionHeaderPanel.xaml.cs [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+                            "https://cwe.mitre.org/data/definitions/{0}.html");
+                }
+            }
+        }
+
+        private void AddLinksAndDisplayPanel(StackPanel panel, string[] linksData, string namePrefix, string urlPatter)
+        {
+            panel.Children.Clear();
+
+            if (linksData != null && linksData.Length > 0)
```
Comment (michelkaporin) [inline Snyk.VisualStudio.Extension/Snyk/VisualStudio/Extension/UI/Toolwindow/DescriptionHeaderPanel.xaml.cs:129]: It's a good programming habit to simplify the nesting of code by doing early returns. This provides much more readable code without the need to scroll to the "else" clause, if "if" is large.
In this case you'd do:

```csharp
if (linksData == null || linksData.Length == 0) {
   panel.Visibility = Visibility.Collapsed;
   return;
}

// the current "if" goes here.
```

File: Snyk.VisualStudio.Extension/Snyk/VisualStudio/Extension/UI/Toolwindow/DescriptionHeaderPanel.xaml.cs [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+            {
+                panel.Visibility = Visibility.Visible;
+
+                foreach (string linkData in linksData)
```
Comment (michelkaporin) [inline Snyk.VisualStudio.Extension/Snyk/VisualStudio/Extension/UI/Toolwindow/DescriptionHeaderPanel.xaml.cs:133]: use `var` in foreach declaration, as well as in the code below.

File: Snyk.VisualStudio.Extension/Snyk/VisualStudio/Extension/UI/Toolwindow/DescriptionHeaderPanel.xaml.cs [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+                    this.AddLinksAndDisplayPanel(
+                            this.cwePanel,
+                            suggestion.Cwe.ToArray(),
+                            "CWE-",
+                            "https://cwe.mitre.org/data/definitions/{0}.html");
+                }
+            }
+        }
+
+        private void AddLinksAndDisplayPanel(StackPanel panel, string[] linksData, string namePrefix, string urlPatter)
```
Comment (michelkaporin) [inline Snyk.VisualStudio.Extension/Snyk/VisualStudio/Extension/UI/Toolwindow/DescriptionHeaderPanel.xaml.cs:125]: `linksData` is too abstract as a parameter name, how being more explicit? e.g. `vulnerabilityId`

Comment (michelkaporin) [reply Snyk.VisualStudio.Extension/Snyk/VisualStudio/Extension/UI/Toolwindow/DescriptionHeaderPanel.xaml.cs:125]: another nitpick: Is it AddLinks**To**DisplayPanel, not AddLinks**And**DisplayPanel?

File: Snyk.Code.Library/Domain/Analysis/Suggestion.cs [hunk 6]
```diff
@@ -77,9 +77,15 @@ public class Suggestion
...
+        public string GetDisplayTitleWithLineNumber() => "Line " + this.Rows.Item1 + ": " + this.GetDisplayTitle();
+
+        /// <summary>
+        /// Get title. If title is null or empty it will return message as title.
+        /// </summary>
+        /// <returns>Title for display.</returns>
         public string GetDisplayTitle()
         {
-            string title = "Line " + this.Rows.Item1 + ": ";
+            string title = string.Empty;
```
[the remark this answers is not in the document]
Comment (aldanchenko) [reply Snyk.Code.Library/Domain/Analysis/Suggestion.cs:88]: Good point! Thanks @michelkaporin

File: Snyk.VisualStudio.Extension/Snyk/VisualStudio/Extension/UI/Toolwindow/DescriptionHeaderPanel.xaml.cs (hunk 5 above)
Comment (aldanchenko) [reply Snyk.VisualStudio.Extension/Snyk/VisualStudio/Extension/UI/Toolwindow/DescriptionHeaderPanel.xaml.cs:125]: Maybe I misunderstood something. This method Add links to panel and display (or hide) this panel according to provided data. I tried to split this logic (add links and display/hide), but it depends on provided data. So by my opinion right now  AddLinks**And**DisplayPanel conveys logic more accurately. But I can change the name.

Comment (aldanchenko) [reply Snyk.VisualStudio.Extension/Snyk/VisualStudio/Extension/UI/Toolwindow/DescriptionHeaderPanel.xaml.cs:125]: @michelkaporin found another approach for this part. Please, check last changes. Method renamed to AddLinksToPanel.

**Review by michelkaporin — APPROVED** [review]
Blazingly reactive amends from your side 😮 Looks good!

**Pull request merged**